### ISSUE TITLE
flipX method doesn't work on a GameObj (v0.5.1)

### ISSUE THREAD

**Issue by perscrew** [replit/kaboom#127]
Hello, I have a weird issue regardind the `flipX`method on a `GameObj`.

My sprite disappeared when I call `player.flipX(false);`.

Here you 'll find a short code example : 

```
k.keyDown('left', () => {
      if (!player.jumpState.active && player.curAnim() !== 'run') {
        player.play('run');
      }
      player.flipX(true); // here the flipX doesn't work
      player.move(-PLAYER_SPEED, 0);
    });

    k.keyRelease('left', () => {
      if (!k.keyIsDown('left') && player.curAnim() !== 'jump') {
        player.play('idle');
      }
    });

    k.keyDown('right', () => {
      if (!player.jumpState.active && player.curAnim() !== 'run') {
        player.play('run');
      }
      player.flipX(false); // here the sprite disapeared
      player.move(PLAYER_SPEED, 0);
    });

    k.keyRelease('right', () => {
      if (!k.keyIsDown('right') && player.curAnim() !== 'jump') {
        player.play('idle');
      }
    });
```

I used the `0.5.1` version hosted on npm. I also tested on a replit and I got the same problem.

I  also checked your `level` example which seems to use the `0.6.0` version, the `flipX` works correctly.

Is it possible to publish the `0.6.0` or `0.7.0` on npm ?

Maybe I miss something ?

**Comment by medavids** [comment]
Hi,
use flipX(-1) or flipX(1) - depending on how do you set your sprite.
for example:
`player.flipX(-1); //flip sprite to left`

**Comment by perscrew** [comment]
Thanks @medavids, it works like a charm !

Is it possible to add `flipX` to type declaration files ? I'm using typescript and this method is interpreted like a key value due to : 

```
export type GameObj = {
  ... // omitted
  [custom: string]: any,
}
```

**Comment by slmjkdbtl** [comment]
It's a typescript problem kaboom is having, fields are dynamically assigned according to the components passed to `add()`

**Comment by perscrew** [comment]
Ok, I took a look at the `kaboom.ts`. 

I see some `typeof` assertion, maybe you could try to use `typescript type guard` to force the inference of the oject passed to the function ? 

If you want I can help to solve this issue with you. 

BTW this issue can be closed. Thanks a lot.

**Comment by slmjkdbtl** [comment]
So right now some type merging problems are solved by #118 , editors should be able to autocomplete / suggest the correct fields now, but have to keep `[custon: string]: any` for a while until I clean up some type looseness in kaboom's codebase that relies on that.
